dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -394,4 +395,36 @@ def _convert_time_dimension(self, load_data_df, project_config):
 
         if time_dim.model.is_time_zone_required_in_geography():
             load_data_df = load_data_df.drop("time_zone")
+
+        load_data_df = self._convert_dataset_time_to_project_time_interval(
+            load_data_df, project_config
+        )
+        return load_data_df
+
+    def _convert_dataset_time_to_project_time_interval(self, load_data_df, project_config):
+        """Shift dataset time to match project time based on TimeIntervalType"""
```

Comment:
If this function stays here the docstring should clarify that load_data_df is the dataset with the time dimension already partially converted to project time. (Assuming I'm reading this function and _convert_time_dimension correctly. Alternatively, maybe this function could be part of TimeDimension / called by time_dim.convert_dataframe?